### PR TITLE
CD-270 update base theme

### PR DESCRIPTION
**Pull request by left23**
This PR adds the v4 base theme and makes the necessary tweaks to the sub theme to accommodate.
It also replaces Gulp with npm scripts.

Run `composer install`

**The main changes are:**

1. Full width header and footer and removal of side margins means the width of the container increases slightly.
2. The body background is now white, no longer light grey.
3. The header height increases due to 6px bottom border, and the header grid, the placement of the mobile menu dropdown and the active menu item bottom border change to accommodate.
4. Sticky footer for anon users
5. Remove classy as base theme https://github.com/UN-OCHA/common_design/commit/378e343b8fbe9192657801ef2f0492374e7d1412
6. Folders have been added to organise templates https://github.com/UN-OCHA/common_design/tree/main/templates. If a template is missing it may have been moved, or it may have come from Classy base theme.
7. Rename and remove variables and introduce CSS custom properties https://github.com/UN-OCHA/common_design/blob/main/sass/cd/_cd-variables.scss and update subtheme accordingly.
8. Update node version and replace gulp-sass - run `nvm use` and `npm i` to update
9. Remove page title related preprocess functions from the subtheme, in favour of base theme. There is config for the base theme settings - run `drush cim`
 
I am writing release notes for v3.0.0 which I will share asap. https://github.com/UN-OCHA/common_design/releases/tag/v3.0.0

I would like to deploy this asap so I can run VRTs on the server. I have run them locally and due to the nature of the changes, they will typically fail. VRT results I can share will allow me point out acceptable visual differences.

URLs for testing https://docs.google.com/spreadsheets/d/1GcqvK2PWuSZbLKEvQQNKtBc0xOx-GzTSbiIdd6HP5lI/edit#gid=1791254445

### CODE REVIEW

**Review by orakili — COMMENTED** [review]
Looks good to me. I tested various pages in different languages etc. and in different browsers (including IE11) and compared with the gho.unocha.org site. The only thing is the question below about the `cd-bleed` on stories.

Edit: there is a follow-up PR to save the node title settings for the subtheme as they are not inherited from the base theme: https://github.com/UN-OCHA/gho-site/pull/225.

File: html/themes/custom/common_design_subtheme/templates/paragraphs/paragraph--story.html.twig [hunk 1]
```diff
@@ -51,6 +52,7 @@
     'gho-story-paragraph',
     'gho-aside',
     'gho-aside--dark',
+    'cd-bleed'
```
[the remark this answers is not in the document]
Comment (left23) [reply html/themes/custom/common_design_subtheme/templates/paragraphs/paragraph--story.html.twig:55]: @orakili I'm aware the CSS custom property technique doesn't work in IE. I thought I had gotten around that by adding https://github.com/UN-OCHA/common_design/blob/main/components/cd-bleed/cd-bleed.css#L45
From a visual display perspective, I think it works in IE with this CSS rule in place.
The inclusion here is primarily for the demo, yes, but we are also trying to cater for full-width elements and this is how I intended to go about it.

**Comment by left23** [comment]
I'm seeing an issue with the Roboto font in the header. It seems the 300 weight is not loading so the OCHA Services and User menu and Language switcher font is very light.
Screenshot from 2021-04-22 15-57-11

**Comment by left23** [comment]
@orakili @rupl 
I've pushed a fix for the Header and Footer font weight. I think what was happening was the Roboto 300 font weight wasn't being included, so the font weight was defaulting to 400 (you can see this on Prod)
When I added the base theme updates, which includes the 300 Roboto, the font-weight is too light. I've set the weight to 400 explicitly for the header and footer, which I think results in what was intended visually.

Please have a look and confirm or adjust.

**Comment by left23** [comment]
@orakili @rupl I'm re-requesting a review with the latest updates. Not much should have changed visually but please double check and span the latest commits.
I'm deploying this to the dev site and I will do VRTs once Jenkins decides to play nice.

**Review by orakili — APPROVED** [review]
Tested locally and looked goo to me.

**Comment by lazysoundsystem** [comment]
Merged as per: https://www.flowdock.com/app/unocha/global-humanitarian-overview/threads/S38_VeDPaNqVpIERCA8DcOuWJIG

**Pull request merged**